src/crypto/symmetric_key.cpp

```diff
@@ -19,4 +19,44 @@ namespace crypto
     return std::make_unique<KeyAesGcm_OpenSSL>(rawKey);
 #endif
   }
+
+  std::vector<uint8_t> aes_gcm_encrypt(
+    const std::vector<uint8_t>& key,
+    std::vector<uint8_t>& plaintext,
+    const std::vector<uint8_t>& iv,
+    const std::vector<uint8_t>& aad)
+  {
+    if (key.size() != 16 && key.size() != 24 && key.size() != 32)
```

Comment:
Can we move this out to a single checking call? I'm uncomfortable with having literals like that in multiple places.

Reply:
Sure, np.